float64 left_depth
float64 center_depth
float64 right_depth
int32 di